src/Controller/LoginController.php

```diff
@@ -14,32 +17,36 @@
 class LoginController extends AbstractController
 {
     /**
-     * @var OpenIdConfigurationProvider
+     * @var OpenIdConfigurationProviderManager
      */
-    private $provider;
+    private $providerManager;
 
-    public function __construct(OpenIdConfigurationProvider $provider)
+    public function __construct(OpenIdConfigurationProviderManager $providerManager)
     {
-        $this->provider = $provider;
+        $this->providerManager = $providerManager;
     }
 
     /**
      * Login method redirecting to authorizer.
      *
+     * @param Request $request
      * @param SessionInterface $session
      * @return RedirectResponse
      * @throws ItkOpenIdConnectException
      */
-    public function login(SessionInterface $session): RedirectResponse
+    public function login(Request $request, SessionInterface $session, string $provider): RedirectResponse
     {
-        $nonce = $this->provider->generateNonce();
-        $state = $this->provider->generateState();
+        $theProvider = $this->providerManager->getProvider($provider);
```

Comment:
Suggest rename:
`$provider` -> `$providerKey`
`$theProvider` -> `$provider`

(This will also match the naming you use in `OpenIdLoginAuthenticator`)

Reply:
I changed this (from what you suggest) to the current names to have the route path be `/login/{provider}` rather than `/login/{providerKey}`. I prefer the former, but I don't like `$theProvider` either.

Reply:
I missed that it is part of the route. Understand your choice. However my preference would still be `/login/{providerKey}` as this is more explicit in regards to what should be provided. I think consistent naming limits confusion and ambiguity in regards to how to use and configure the bundle. 

This would be useful in the readme, we could expand the config section to make it clearer how the `providerKey` is defined
```
  openid_providers:
    # Define one or more providers
    [providerKey]:
      options:
        metadata_url: '%env(ADMIN_OIDC_METADATA_URL)%'
        client_id: '%env(ADMIN_OIDC_CLIENT_ID)%'
        client_secret: '%env(ADMIN_OIDC_CLIENT_SECRET)%'
        # Specify redirect URI
        redirect_uri: '%env(ADMIN_OIDC_REDIRECT_URI)%'```

Reply:
Fixed.